tradebot/data/freshness.py

```
@@ -0,0 +1,378 @@
+        # Create market open datetime for the given date
+        market_open = dt.replace(
+            hour=self.market_open_time.hour,
+            minute=self.market_open_time.minute,
+            second=0,
+            microsecond=0
+        )
+        
+        return market_open
+    
+    def is_data_fresh(self, symbol: str, start_date: str, end_date: str, 
+                     threshold_minutes: Optional[int] = None) -> bool:
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)